gaphor/UML/actions/partitionpage.py

```diff
@@ -23,60 +23,74 @@ class PartitionPropertyPage(PropertyPageBase):
     def __init__(self, item: PartitionItem):
         super().__init__()
         self.item = item
-        self.list_store = Gtk.ListStore(int, str)
 
     def construct(self):
         """Creates the Partition Property Page."""
         builder = new_builder(
             "partition-editor",
+            "num-partitions-adjustment",
             signals={"partitions-changed": (self._on_num_partitions_changed,)},
         )
 
         num_partitions = builder.get_object("num-partitions")
-        adjustment = Gtk.Adjustment(
-            value=len(self.item.partition),
-            lower=1,
-            upper=10,
-            step_increment=1,
-            page_increment=5,
-        )
-        num_partitions.set_adjustment(adjustment)
+        num_partitions.set_value(len(self.item.partition))
+
+        self.partitions = builder.get_object("partitions")
+        for partition in self.item.partition:
+            self.partitions.pack_start(
+                self.construct_partition(partition), expand=False, fill=False, padding=0
+            )
 
-        treeview = builder.get_object("partition-treeview")
-        treeview.set_model(self.list_store)
+        return builder.get_object("partition-editor")
 
-        renderer_text = Gtk.CellRendererText()
-        treeview.append_column(
-            Gtk.TreeViewColumn(title="#", cell_renderer=renderer_text, text=0)
+    def construct_partition(self, partition):
+        builder = new_builder(
+            "partition",
+            signals={
+                "partition-name-changed": (self._on_partition_name_changed, partition),
+                "partition-type-changed": (self._on_partition_type_changed, partition),
+            },
         )
 
-        renderer_editable_text = Gtk.CellRendererText()
-        renderer_editable_text.set_property("editable", True)
+        builder.get_object("partition-name").set_text(partition.name or "")
 
-        treeview.append_column(
-            Gtk.TreeViewColumn(
-                title=gettext("Name"), cell_renderer=renderer_editable_text, text=1
-            )
-        )
+        combo = builder.get_object("partition-type")
+        for c in self.item.model.select(UML.Classifier):
```

Comment:
I think we should limit this to Blocks and Classes. Being able to select an Activity to allocate an Activity Partition to isn't quite right, since the partition should be allocated to a structural element, not behavioral.

Reply:
I think it makes sense to stick with classifiers, since that includes interfaces too. We can exclude behaviors.

Reply:
This looks good now :+1: